machine_learning_hep/utilities_plot.py

```diff
@@ -24,25 +24,61 @@
 import numpy as np
 from root_numpy import fill_hist # pylint: disable=import-error, no-name-in-module
 # pylint: disable=import-error, no-name-in-module
-from ROOT import TH2F, TFile, TH1, TH3F, TGraphAsymmErrors
+from ROOT import TH1F, TH2F, TFile, TH1, TH3F, TGraphAsymmErrors
 from ROOT import TPad, TCanvas, TLegend, kBlack, kGreen, kRed, kBlue, kWhite
 from ROOT import gStyle, gROOT
 from machine_learning_hep.io import parse_yaml, dump_yaml_from_dict
 from machine_learning_hep.logger import get_logger
 
 def buildarray(listnumber):
+    """
+    Build an array out of a list, useful for histogram binning
+    """
     arraynumber = array('d', listnumber)
     return arraynumber
 
-def makefill2dhist(df_, titlehist, arrayx, arrayy, nvar1, nvar2):
+def buildbinning(nbinsx, xlow, xup):
     """
-    Create a TH2F histogram and fill it with two variables from a dataframe.
+    Build a list for binning out of bin limits and number of bins
     """
-    lenx = len(arrayx) - 1
-    leny = len(arrayy) - 1
+    listnumber = [xlow + (xup - xlow) / nbinsx * i for i in range(nbinsx + 1)]
+    return buildarray(listnumber)
 
-    histo = TH2F(titlehist, titlehist, lenx, arrayx, leny, arrayy)
+def buildhisto(h_name, h_tit, arrayx, arrayy=None, arrayz=None):
+    """
+    Create a histogram of size 1D, 2D, 3D, depending on the number of arguments given
+    """
+    histo = None
+    def binning(binning_array):
+        return len(binning_array) - 1, binning_array
+    if arrayz:
+        histo = TH3F(h_name, h_tit, *binning(arrayx), *binning(arrayy), *binning(arrayz))
```

Comment:
hi @njacazio, how is it * operation called?

Reply:
@ginnocen https://docs.python.org/3.6/tutorial/controlflow.html#unpacking-argument-lists

Reply:
Thank you @ginnocen